rubygem/lib/zeus.rb

```diff
@@ -192,28 +194,18 @@ def report_error_to_master(local, error)
       local.write str
     end
 
-    def run_action(socket, identifier, feature_pipe_w)
+    def run_action(socket, identifier)
       # Now we run the action and report its success/fail status to the master.
-      features, err = Zeus::LoadTracking.features_loaded_by do
-        plan.after_fork unless identifier == :boot
-        plan.send(identifier)
-      end
-
-      if err
-        # If we received an error, report features to the master syncronously.
-        # We need to do this before reporting the error to the master
-        # otherwise it will kill us before we can report features.
-        begin
-          notify_features(feature_pipe_w, features)
-          feature_pipe_w.close
-        ensure
-          report_error_to_master(socket, err)
+      begin
+        Zeus::LoadTracking.track_features_loaded_by do
+          plan.after_fork unless identifier == :boot
+          plan.send(identifier)
         end
-      else
-        # If we booted successfully, report features in a new thread
-        # so we can immediately begin listening for commands.
+
         socket.write "R:OK\0"
-        Thread.new { notify_features(feature_pipe_w, features) }
+      rescue => err
+        report_error_to_master(socket, err)
+        raise
```

Comment:
It shouldn't. Zeus doesn't connect any log output for child nodes by default. We do log output from Hera but I think we redirect these errors in a way that'll avoid the double-spew. I'm planning to cut a new release today, I'll install it for myself first and test the error behavior a bit more.